### PR TITLE
Basic version of goto-diff tool

### PR DESCRIPTION
**Pull request by peterschrammel**
basic syntactic diffing

requires pull request #62

### CODE REVIEW

File: src/cbmc/cbmc_parse_options.cpp [hunk 1]
```diff
@@ -551,7 +552,12 @@ int cbmc_parse_optionst::doit()
     
     // Entry point will have been set before and function pointers removed
     status() << "Removing Unused Functions" << eom;
-    remove_unused_functions(goto_functions, ui_message_handler);
+
+    //ENHANCE: this should be done somewhere else
+    if(config.main == "") //no function cmd line option given
```
Comment (kroening) [inline src/cbmc/cbmc_parse_options.cpp]: It's unclear to me why this needs to be done here?

File: src/cbmc/bmc.cpp [hunk 2]
```diff
@@ -223,6 +224,14 @@ void bmct::report_success()
       std::cout << "\n";
     }
     break;
+
+  case ui_message_handlert::JSON_UI:
```
Comment (kroening) [inline src/cbmc/bmc.cpp]: Could all the JSON_UI stuff go into a separate pull request?

Comment (peterschrammel) [inline src/cbmc/bmc.cpp]: #62 and #65 are clean now. You have to merge #62 before #65

**Comment by tautschnig** [comment]
@peterschrammel it may make sense to merge the first 4 patches of my series into "goto-diff initial" or one of the later ones as appropriate.

**Comment by peterschrammel** [comment]
Yes. And it needs rebasing too. I'll clean it up as soon as #62 is merged.

**Comment by tautschnig** [comment]
@peterschrammel Thanks for cleaning up #62 and @kroening thanks for merging! Looking forward to getting this one refined, I'll then see what further patches on top of this one I might already have ready for submission.

**Comment by peterschrammel** [comment]
@tautschnig, maybe it makes sense to pull your patches into https://github.com/peterschrammel/cbmc/tree/goto-diff right away as they are likely to be invalidated by my clean-up.

**Comment by tautschnig** [comment]
@peterschrammel Apologies for taking a while to get this done, but I've now pushed all the further and create a new pull request with your fork (but you will have noticed anyway).
